### PR TITLE
Add lndmon and refactor lnd + subcharts

### PR DESCRIPTION
**Pull request by krtk6160**

### CODE REVIEW

File: charts/lnd/charts/lndmon/values.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: v0.2.0
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
```
Comment (nicolasburtey) [inline charts/lnd/charts/lndmon/values.yaml:19]: lndmon doesn't need a service account?

File: charts/lnd/charts/loop/Chart.yaml [hunk 2]
```diff
@@ -15,9 +15,10 @@ type: application
...
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.1.3
+version: 0.1.4
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
-appVersion: 0.11.1
+# It is recommended to use it with quotes.
```
Comment (nicolasburtey) [inline charts/lnd/charts/loop/Chart.yaml:23]: why is that?

File: charts/lnd/charts/rtl/values.yaml [hunk 3]
```diff
@@ -1,22 +1,69 @@
...
-  type: ClusterIP
+podAnnotations: {}
 
-podSecurityContext: 
-  fsGroup: 2000
+podSecurityContext: {}
+  # fsGroup: 2000
 
-securityContext:
+securityContext: {}
```
Comment (nicolasburtey) [inline charts/lnd/charts/rtl/values.yaml:24]: why deactivate non root access?

File: charts/lnd/templates/wallet-init-hook.yaml [hunk 4]
```diff
@@ -14,7 +15,7 @@ spec:
     image: "boxboat/kubectl"
     imagePullPolicy: IfNotPresent
     command: ['/bin/sh']
-    args: ['-c','apk update; apk add expect; ./walletInit.exp {{.Values.global.network}} $LND_PASS {{ include "lnd.fullname" . }}-0;']
+    args: ['-c', 'apk update; apk add expect; kubectl wait --for=condition=ready --timeout=150s pod {{ include "lnd.fullname" . }}-0; ./walletInit.exp {{.Values.global.network}} $LND_PASS {{ include "lnd.fullname" . }}-0;']
```
Comment (nicolasburtey) [inline charts/lnd/templates/wallet-init-hook.yaml:18]: the `.exp` is confusing to me. what is that supposed to mean?

File: charts/lnd/templates/wallet-init-hook.yaml [hunk 5]
```diff
@@ -6,6 +6,7 @@ metadata:
   annotations:
     "helm.sh/hook": "post-install"
 spec:
+  # FIXME: should not run as root
   securityContext:
     runAsUser: 0
```
Comment (nicolasburtey) [inline charts/lnd/templates/wallet-init-hook.yaml:11]: we could run it as non root with the new image I think?

**Review by nicolasburtey — COMMENTED** [review]
solid PR. great to see this refactoring. couple of questions.

File: charts/lnd/charts/loop/Chart.yaml (hunk 2 above)
Comment (krtk6160) [reply charts/lnd/charts/loop/Chart.yaml:23]: Not sure. If you do `helm create xyz`, the Chart.yaml will be generated with this comment i.e. it's not a custom comment

File: charts/lnd/templates/wallet-init-hook.yaml (hunk 4 above)
Comment (krtk6160) [reply charts/lnd/templates/wallet-init-hook.yaml:18]: Any extension can be used, but tt's a convention for expect scripts to use `.exp` extension. More info [here](https://docs.oracle.com/cd/E35328_01/E35336/html/vmcli-script.html). 

It's a way of self documenting to make it clear that it's not a bash script.

File: charts/lnd/charts/rtl/values.yaml (hunk 3 above)
Comment (krtk6160) [reply charts/lnd/charts/rtl/values.yaml:24]: Didn't realize it. Fixed.

File: charts/lnd/charts/lndmon/values.yaml (hunk 1 above)
Comment (krtk6160) [reply charts/lnd/charts/lndmon/values.yaml:19]: Fixed

File: charts/lnd/templates/wallet-init-hook.yaml (hunk 5 above)
Comment (krtk6160) [reply charts/lnd/templates/wallet-init-hook.yaml:11]: I tried it, didn't work. This hook actually isn't related to the lnd image at all, it just runs a bash script. The `expect` package needs to be installed, and for that we need to run this as a root user.

**Pull request merged**